pytest_mergify/__init__.py

```diff
@@ -85,14 +93,68 @@ def tracer(self) -> typing.Optional[opentelemetry.trace.Tracer]:
         return self.mergify_tracer.tracer
 
     def pytest_sessionstart(self, session: _pytest.main.Session) -> None:
+        self.has_error = False
+
         if self.tracer:
             self.session_span = self.tracer.start_span(
                 "pytest session start",
                 attributes={
                     "test.scope": "session",
                 },
             )
-        self.has_error = False
+
+        if not utils.is_in_ci():
+            return
+
+        if self.mergify_tracer.repo_name is None:
+            self.quarantine_warning_logs.append(
+                "No valid repository name found, skipping CI Insights Quarantine setup"
+            )
+            return
+
+        # Query all the quarantined tests
+        try:
+            owner, repository = self.mergify_tracer.repo_name.split("/")
+        except ValueError:
+            self.quarantine_warning_logs.append(
+                f"Repository name '{self.mergify_tracer.repo_name}' has an unexpected format (expected 'owner/repository'), skipping CI Insights Quarantine setup"
+            )
+            return
+
+        branch_name = utils.get_branch_name_for_quarantine()
+        if branch_name is None:
+            self.quarantine_warning_logs.append(
+                "Could not find a branch name, which is required to use CI Insights Quarantine",
```

Comment:
nitpic, keeping logs consistent 
```suggestion
                "No valid branch name found, skipping CI Insights Quarantine setup"
```